packages/pwa-kit-create-app/EXTENSIONS.md

```diff
@@ -0,0 +1,400 @@
+
+1. **Validates Project Structure**
+
+    - Confirms required directories exist (e.g., `components`, `pages`)
+    - Verifies target integration files are present
+    - Checks for necessary dependencies
+    - Adds new UI components to page output / default React component return block
+
+2. **Clones Source Repository**
+
+    ```bash
+    mkdir /tmp/pwa-kit-tmp
+    git clone -b develop https://github.com/SalesforceCommerceCloud/pwa-kit /tmp/pwa-kit-tmp
+    ```
+
+3. **Copies Extension Components**
+
+    ```bash
+    # Extension-specific components and assets are copied to the project
+    cp -r /tmp/pwa-kit-tmp/packages/template-chakra-storefront/components/[extension-name] \
+          packages/template-chakra-storefront/components/
+    ```
+
+4. **Updates Application Code**
+   The system applies automated code changes to integrate the extension:
+
+    **Import Statements:**
+
+    ```javascript
+    // Required dependencies are added
+    import loadable from '@loadable/component'
+    import SomeExtensionDependency from 'package-name'
+    ```
+
+    **Component Declarations:**
+
+    ```javascript
+    // Extension components are loaded dynamically
+    const ExtensionComponent = loadable(() => import('../../components/[extension-name]'))
+    ```
+
+    **Integration Points:**
+
+    ```javascript
+    // Extensions are integrated at predefined locations
+    return (
+        <Box data-testid="page" layerStyle="page">
+            <Seo {...seoProps} />
+
+            {/* Extension components added at designated integration points */}
+            <ExtensionComponent />
+
+            <ExistingComponent>{/* Existing application content */}</ExistingComponent>
+        </Box>
+    )
+    ```
+
+5. **Cleanup**
+    ```bash
+    rm -rf /tmp/pwa-kit-tmp
+    ```
+
+### Installation Result
+
+After installation, extensions typically provide:
+
+-   **New Components** - React components added to designated pages
+-   **Enhanced Functionality** - Additional features integrated into the storefront
+-   **Configuration Options** - Customizable settings and parameters
+-   **Styling Integration** - Components that match the existing design system
+
+## Project Structure
+
+```
+packages/template-chakra-storefront/
+├── components/
+│   ├── [extension-name]/             # ✅ Installed extension components
+│   │   ├── index.jsx                 # Main extension component
+│   │   └── [additional-files]        # Supporting files and assets
+│   ├── hero/
+│   ├── seo/
+│   └── ...
+├── pages/
+│   ├── home/
+│   │   └── index.jsx                 # ✅ Updated with extension integration
+│   └── ...
+└── ...
+```
+
+## Extension Selection and Tree-Shaking
+
+### Project Generation Process
+
+When creating a new PWA Kit project using `npx @salesforce/pwa-kit-create-app`, developers can select which extensions to include in their storefront. This selection process ensures that only the desired functionality is included in the final project.
```

Comment:
I was  not able to run the generator in v4-redux/demo1 and next branch. Is this still under development?